cpp/tests/io/json/json_test.cpp

```diff
@@ -2776,4 +2776,47 @@ TEST_F(JsonReaderTest, JSONMixedTypeChildren)
   }
 }
 
+TEST_F(JsonReaderTest, JsonDtypeSchema)
+{
+  std::string data = R"(
+    {"a": 1, "b": {"0": "abc", "1": ["a", "b"]}, "c": true}
+    {"a": 1, "b": {"0": "abc"          }, "c": false}
+    {"a": 1, "b": {"0": "lolol  "}, "c": true}
+    )";
+
+  std::map<std::string, cudf::io::schema_element> dtype_schema{{"c", {data_type{type_id::STRING}}},
+                                                               {"b", {data_type{type_id::STRING}}},
+                                                               {"a", {dtype<double>()}}};
+  cudf::io::json_reader_options in_options =
+    cudf::io::json_reader_options::builder(cudf::io::source_info{data.data(), data.size()})
+      .dtypes(dtype_schema)
+      .mixed_types_as_string(true)
```

Comment:
do we still need `mixed_types_as_string`, given that we specified the data types?

Reply:
No, we don't. I have replaced this with the requirement that `prune_columns` need to be enabled for non-mixed type nested columns to be parsed as string.

Reply:
I have added a check for the unsupported case - when `is_strict_nested_boundaries` is false, but we want to normalize whitespaces in the nested column forced as string.